src/test/test_dir.c

```diff
@@ -3830,6 +3864,83 @@ mock_get_options(void)
   return mock_options;
 }
 
+/**
+ * Test dirauth_get_b64_digest_bw_file.
+ * This function should be near the other bwauth functions, but it needs
+ * mock_get_options, that is only defined here.
+ */
+
+static void
+test_dir_bwauth_bw_file_digest256(void *arg)
+{
+  (void)arg;
+  const char *content =
+    "1541171221\n"
+    "node_id=$68A483E05A2ABDCA6DA5A3EF8DB5177638A27F80 "
+    "master_key_ed25519=YaqV4vbvPYKucElk297eVdNArDz9HtIwUoIeo0+cVIpQ "
+    "bw=760 nick=Test time=2018-05-08T16:13:26\n";
+
+  char *fname = tor_strdup(get_fname("V3BandwidthsFile"));
+  /* Initialize to a wrong digest. */
+  uint8_t digest[DIGEST256_LEN] = "01234567890123456789abcdefghijkl";
+
+  /* Digest of an empty string */
+  char digest_empty_str[DIGEST256_LEN] = "01234567890123456789abcdefghijkl";
+  crypto_digest256(digest_empty_str, "", 0, DIGEST_SHA256);
+
+  /* Digest of the content */
+  char digest_expected2[DIGEST256_LEN] = "01234567890123456789abcdefghijkl";
+  crypto_digest256(digest_expected2, content, sizeof(content), DIGEST_SHA256);
+  /* Logs 1DC64B095F76C89980B23C99D06B749D6B46DAA9C270E4FA1E027605CBC02A39 */
+  log_debug(LD_DIR, "%s", hex_str(digest_expected2, DIGEST256_LEN));
+  /* Using sha256sum in command line a file with the content gives
+   * ecaa36314fc835d3269ca304d2f9be3c87cbd2c068faee0c2b8aabfe538bef8c */
+
+  /* Digest of the content */
+  uint8_t digest_expected[DIGEST256_LEN] = "01234567890123456789abcdefghijkl";
+  crypto_digest_t *digest_tmp = crypto_digest256_new(DIGEST_SHA256);
+  crypto_digest_add_bytes(digest_tmp, content, strlen(content));
+  crypto_digest_get_digest(digest_tmp, (char *) digest_expected,
+                           DIGEST256_LEN);
+  /* Logs 27EF4B9BB513E056164885E038970088859B1C17A2525FB472610E2A9C3B1679 */
+  log_debug(LD_DIR, "%s",
+            hex_str((const char *) digest_expected, DIGEST256_LEN));
+
+  /* When there is not a bandwidth file configured */
```

Comment:
```suggestion
  /* When the bandwidth file can not be found. */
```

Reply:
When a bandwidth file isn't configured, the spec says that there should not be a hash in the vote. So this comment is confusing.